src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs

```diff
@@ -204,31 +206,61 @@ private static bool TryGetSuffixLength(StringSegment subType, out int suffixLeng
         /// <summary>
         /// Gets the subtype of the <see cref="MediaType"/>.
         /// </summary>
+        /// <example>
+        /// For the media type <c>"application/vnd.example+json"</c>, the property gives the value
+        /// <c>"vnd.example+json"</c>.
+        /// </example>
         public StringSegment SubType { get; private set; }
 
         /// <summary>
         /// Gets the subtype of the <see cref="MediaType"/>, excluding any structured syntax suffix.
         /// </summary>
+        /// <example>
+        /// For the media type <c>"application/vnd.example+json"</c>, the property gives the value
+        /// <c>"vnd.example"</c>.
+        /// </example>
         public StringSegment SubTypeWithoutSuffix { get; private set; }
 
         /// <summary>
         /// Gets the structured syntax suffix of the <see cref="MediaType"/> if it has one.
         /// </summary>
+        /// <example>
+        /// For the media type <c>"application/vnd.example+json"</c>, the property gives the value
+        /// <c>"json"</c>.
+        /// </example>
         public StringSegment SubTypeSuffix { get; private set; }
 
         /// <summary>
         /// Gets whether this <see cref="MediaType"/> matches all subtypes.
         /// </summary>
+        /// <example>
+        /// For the media type <c>"application/*"</c>, this property is <c>true</c>.
+        /// </example>
+        /// <example>
+        /// For the media type <c>"application/json"</c>, this property is <c>false</c>.
+        /// </example>
         public bool MatchesAllSubTypes => SubType.Equals("*", StringComparison.OrdinalIgnoreCase);
 
         /// <summary>
         /// Gets whether this <see cref="MediaType"/> matches all subtypes, ignoring any structured syntax suffix.
         /// </summary>
+        /// <example>
+        /// For the media type <c>"application/*+json"</c>, this property is <c>true</c>.
+        /// </example>
+        /// <example>
+        /// For the media type <c>"application/vnd.example+json"</c>, this property is <c>false</c>.
+        /// </example>
         public bool MatchesAllSubTypesWithoutSuffix => SubTypeWithoutSuffix.Equals("*", StringComparison.OrdinalIgnoreCase);
 
         /// <summary>
         /// Gets whether this <see cref="MediaType"/> matches all nonempty structured syntax suffixes.
         /// </summary>
+        /// <example>
+        /// For the media type <c>"application/vnd.example+*"</c>, this property is <c>true</c>.
+        /// </example>
+        /// <example>
+        /// For the media type <c>"application/vnd.example+json"</c>, this property is <c>false</c>.
+        /// </example>
         public bool MatchesAllNonemptySubTypeSuffixes => SubTypeSuffix.Equals("*", StringComparison.OrdinalIgnoreCase);
```

Comment:
I still don't understand why this is needed. This isn't defined by any spec that I've seen and no one has asked for it.